features/messages/impl/src/test/kotlin/io/element/android/features/messages/textcomposer/MessageComposerPresenterTest.kt

```diff
@@ -342,13 +286,8 @@ class MessageComposerPresenterTest {
 
     @Test
     fun `present - Pick media from gallery`() = runTest {
-        val fakeMatrixRoom = FakeMatrixRoom()
-        val presenter = MessageComposerPresenter(
-            this,
-            fakeMatrixRoom,
-            pickerProvider,
-            featureFlagService,
-        )
+        val presenter = createPresenter(this)
+        pickerProvider.givenMimeType(MimeTypes.Images)
```

Comment:
I see a risk of configuring the global `pickerProvider` in a test. Test ordering is not guarantee.

Maybe reset `pickerProvider` in `fun createPresenter`?

Reply:
What do you mean by 'the global pickerProvider'? AFAIK this instance will be recreated for every test case. In fact, I just added some logs in its constructor to make sure that was the case and each test result had the log with a different `this.hashCode()`.